plugin.cpp

```diff
@@ -418,8 +508,19 @@ void plugin_reconfigure(PLUGIN_HANDLE *handle, const string& newConfig)
 					return;
 				}
 			}
+			else if(actionStr == "remove")
+			{
+				actn = action::REMOVE;
+
+				if (iter->HasMember("datapoint"))
+                                        datapoint = (*iter)["datapoint"].GetString();
+
+				if (iter->HasMember("type"))
+                                        dpType = (*iter)["type"].GetString();
+
+			}
 			Logger::getLogger()->info("Parse asset filter config, Adding to assetFilterConfig map: {%s, %d, %s}", asset_name.c_str(), actn, new_asset_name.c_str());
-			(*newInfo->assetFilterConfig)[asset_name] = {actn, new_asset_name, dpmap};
+			(*newInfo->assetFilterConfig)[asset_name] = {actn, new_asset_name, dpmap, datapoint, dpType};
```

Comment:
at line#525 : tmp variable  is not used anywhere. Can it be removed ?
at line#527 : memory for existing pointer will be leaked here. Also instead of pointer assignment can we use std::move() here ?
Only two files are used used from newInfo variable. So instead of allocating memory for entire structure, it's better to allocate memory just for required one. Also who will delete newInfo pointer ?

Reply:
not related to this jira code changes, out of scope